packages/bot-skeleton/src/scratch/utils/index.js

```diff
@@ -112,23 +123,34 @@ export const load = (block_string, drop_event, showIncompatibleStrategyDialog) =
     }
 
     try {
-        const event_group = `dbot-load${Date.now()}`;
+        const is_collection = xml.hasAttribute('collection') && xml.getAttribute('collection') === 'true';
+        const event_group = is_collection ? 'load_collection' : `dbot-load${Date.now()}`;
+        const workspace = preview_workspace || Blockly.derivWorkspace;
```

Comment:
Same here, better to pass in the workspace as an argument than falling back to a global.

Reply:
I was thinking is not necessary for everyplace that called this function to pass in extra parameter because most of them use `Blockly.derivWorkspace`. But yeah sure.